### PR TITLE
v5.31 changelog

### PR DESCRIPTION
**Pull request by amyblais**

### CODE REVIEW

File: source/administration/changelog.md [hunk 1]
```diff
@@ -4,6 +4,69 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+### Go Version
+ - 5.31 is built with Go ``1.14.6``.
+
+### Database Changes
+ - 
+
+### Websocket Event Changes
+ - Added new websocket events ``thread_updated``, ``thread_follow_changed``, and ``thread_read_changed``.
+
+### API Changes
```
Comment (amyblais) [inline source/administration/changelog.md:47]: TBD by dev

File: source/administration/changelog.md [hunk 2]
```diff
@@ -4,6 +4,69 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ 
+### Known Issues
+ - **Discard Changes** confirmation is not displayed when an admin adds people in the **System Roles** System Console page and clicks elsewhere before saving the changes.
```
Comment (cwarnermm) [inline source/administration/changelog.md:52]:
```suggestion
 - **Discard Changes** confirmation is not displayed when an admin adds people on the **System Roles** System Console page and clicks elsewhere before saving the changes.
```

File: source/administration/changelog.md [hunk 3]
```diff
@@ -4,6 +4,69 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Error text is missing when the team name is left blank on the team creation page.
+ - Line numbers do not line up with the text on code file previews.
```
Comment (cwarnermm) [inline source/administration/changelog.md:54]: This is listed as a bug fix above, and can likely be removed as a known issue.

File: source/administration/changelog.md [hunk 4]
```diff
@@ -4,6 +4,69 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Posts created by bots containing attachments sometimes appear as repeated until the user refreshes the page.
+ - Emoji counter in the center channel doesn't always update immediately when a reaction is added in the right-hand side.
+ - A JavaScript error may appear in some cases when dismissing the new messages toast while scrolled up in the right-hand side.
+ - The Admin Filter option is not disabled in AD/LDAP page for admin roles with ``sysconsole_write_authentication`` permission.
```
Comment (cwarnermm) [inline source/administration/changelog.md:58]:
```suggestion
 - The Admin Filter option is not disabled on the AD/LDAP page for admin roles with the ``sysconsole_write_authentication`` permission.
```

File: source/administration/changelog.md [hunk 5]
```diff
@@ -4,6 +4,69 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
 
+## Release v5.31 - [ESR](https://docs.mattermost.com/administration/extended-support-release.html)
+
+### Highlights
+ - Pre-packaged and pre-installed the Incident Management v1.2.0.
+
+### Improvements
+
+#### User Interface (UI)
+ - Added the ability to mute categories with the experimental sidebar feature.
```
Comment (cwarnermm) [inline source/administration/changelog.md:17]:
```suggestion
 - Added the ability to mute categories with the experimental channel sidebar feature.
```

**Review by cwarnermm — COMMENTED** [review]
Comments/suggestions provided inline.

**Comment by mkraft** [comment]
@amyblais Here are the release notes:

- Added new Shared column to the Channels table. ([#16568](https://github.com/mattermost/mattermost-server/pull/16568), [@wiggin77](https://github.com/wiggin77))
- Added the logic to generate a new renewal link that could be used to start the renewal process ([#16539](https://github.com/mattermost/mattermost-server/pull/16539), [@ethervoid](https://github.com/ethervoid))
- Adds a mattermost command to initialize the database ([#16489](https://github.com/mattermost/mattermost-server/pull/16489), [@mgdelacroix](https://github.com/mgdelacroix))
- Automatic installation and enable of plugins using feature flags. ([#16451](https://github.com/mattermost/mattermost-server/pull/16451), [@crspeller](https://github.com/crspeller))
- Bumped the Go version to 1.15.5. ([#16314](https://github.com/mattermost/mattermost-server/pull/16314), [@metanerd](https://github.com/metanerd))
- Log levels were updated so that 4xx and 501 errors are no longer logged at ERROR level. ([#16612](https://github.com/mattermost/mattermost-server/pull/16612), [@isacikgoz](https://github.com/isacikgoz))
- Pre-package Incident Management v1.2.0. ([#16576](https://github.com/mattermost/mattermost-server/pull/16576), [@lieut-data](https://github.com/lieut-data))

**Review by wiersgallak — APPROVED** [review]

**Comment by mkraft** [comment]
> @amyblais Here are the release notes:
> 
> * Added new Shared column to the Channels table. ([#16568](https://github.com/mattermost/mattermost-server/pull/16568), [@wiggin77](https://github.com/wiggin77))
> * Added the logic to generate a new renewal link that could be used to start the renewal process ([#16539](https://github.com/mattermost/mattermost-server/pull/16539), [@ethervoid](https://github.com/ethervoid))
> * Adds a mattermost command to initialize the database ([#16489](https://github.com/mattermost/mattermost-server/pull/16489), [@mgdelacroix](https://github.com/mgdelacroix))
> * Automatic installation and enable of plugins using feature flags. ([#16451](https://github.com/mattermost/mattermost-server/pull/16451), [@crspeller](https://github.com/crspeller))
> * Bumped the Go version to 1.15.5. ([#16314](https://github.com/mattermost/mattermost-server/pull/16314), [@metanerd](https://github.com/metanerd))
> * Log levels were updated so that 4xx and 501 errors are no longer logged at ERROR level. ([#16612](https://github.com/mattermost/mattermost-server/pull/16612), [@isacikgoz](https://github.com/isacikgoz))
> * Pre-package Incident Management v1.2.0. ([#16576](https://github.com/mattermost/mattermost-server/pull/16576), [@lieut-data](https://github.com/lieut-data))
> * The minimum required Postgres version is now 10.0. Mattermost will now fail to start if the Postgres version is below that. ([#16597](https://github.com/mattermost/mattermost-server/pull/16597), [@agnivade](https://github.com/agnivade))

@amyblais I just updated this to add one more release note.

**Comment by wiggin77** [comment]
> * Added new Shared column to the Channels table. ([#16568](https://github.com/mattermost/mattermost-server/pull/16568), [@wiggin77](https://github.com/wiggin77))

Is this changlelog for 5.31?  My understanding is that the new column is 5.32  @mkraft

**Comment by mkraft** [comment]
> > * Added new Shared column to the Channels table. ([#16568](https://github.com/mattermost/mattermost-server/pull/16568), [@wiggin77](https://github.com/wiggin77))
> 
> Is this changlelog for 5.31? My understanding is that the new column is 5.32 @mkraft

Apparently I used the wrong CLI parameters pulling the release notes from the PRs. The above should be the correct list. @amyblais please note the change.

**Comment by amyblais** [comment]
@mkraft I got a notification that you pinged me, but I don't see the updated list. Perhaps you accidentally deleted your comment?

**Comment by mkraft** [comment]
# API

New `POST /api/v4/cloud/webhook` endpoint.

# Database

No changes.

# Websocket

New events:

* `thread_updated`
* `thread_follow_changed`
* `thread_read_changed`

# Config

New `ServiceSettings.CollapsedThreads` setting.

**Comment by mkraft** [comment]
> @mkraft I got a notification that you pinged me, but I don't see the updated list. Perhaps you accidentally deleted your comment?

Twice I posted incorrect release notes. Looks like I'll need to make some tweaks to the auto release notes process. These are the correct notes: https://github.com/mattermost/docs/pull/4192#issuecomment-759097271

**Review by mkraft — APPROVED** [review]

File: source/administration/changelog.md [hunk 6]
```diff
@@ -4,6 +4,72 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
 Also see [changelog in progress](https://bit.ly/2nK3cVf) for the next release.
 
+## Release v5.31 - [ESR](https://docs.mattermost.com/administration/extended-support-release.html)
+
+**Release Day: 2021-01-16**
+
+### Highlights
+
+#### Improved status updates for [Incident Management (E20 Edition)](https://mattermost.com/incident-management/)
+ - Pre-packaged and pre-installed the Mattermost Incident Management v1.2.0, which enables incident responders to more easily inform stakeholders of incident status updates
```
Comment (cwarnermm) [inline source/administration/changelog.md:14]:
```suggestion
 - Pre-packaged and pre-installed the Mattermost Incident Management v1.2.0, which enables incident responders to more easily inform stakeholders of incident status updates.
```

File: source/administration/changelog.md [hunk 7]
```diff
@@ -4,6 +4,72 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+
+### Improvements
+
+#### User Interface (UI)
+ - Added the ability to mute categories with the experimental channel sidebar feature.
+ - Added support for multi-selection of channels when dragging and dropping between channels in the experimental sidebar feature.
+ - Group messages are now returned in the channel switcher when only first names are typed.
+ - Issuing ``/dnd`` consecutively no longer unexpectedly toggles the user status between “Do Not Disturb” and “Online”, and will only set the user’s status to “Do Not Disturb”.
```
Comment (cwarnermm) [inline source/administration/changelog.md:22]:
```suggestion
 - Issuing ``/dnd`` consecutively no longer unexpectedly toggles the user's status between “Do Not Disturb” and “Online”, and will only set the user’s status to “Do Not Disturb”.
```

File: source/administration/changelog.md [hunk 8]
```diff
@@ -4,6 +4,72 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Fixed an issue where line numbers did not line up with the text on code file previews.
+ - Fixed an issue where the threshold from the bottom of the screen was sometimes not respected for received messages.
+ - Fixed an issue where desktop notifications got sent for every message posted in a Direct Message channel.
+
+### config.json
+Multiple setting options were added to ``config.json``. Below is a list of the additions and their default values on install. The settings can be modified in ``config.json``, or the System Console when available.
+
+#### Changes to Team Edition and Enterprise Edition:
+ - Under ``ServiceSettings`` in config.json:
+    - Added ``CollapsedThreads`` for collapsed reply threads feature.
```
Comment (cwarnermm) [inline source/administration/changelog.md:42]:
```suggestion
    - Added ``CollapsedThreads`` for the collapsed reply threads feature.
```

File: source/administration/changelog.md [hunk 9]
```diff
@@ -4,6 +4,72 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - 5.31 is built with Go ``1.14.6``.
+
+### API Changes
+ - Added a new ``POST /api/v4/cloud/webhook`` endpoint.
+
+### Websocket Event Changes
+ - Added new websocket events ``thread_updated``, ``thread_follow_changed``, and ``thread_read_changed``.
+ 
+### Known Issues
+ - **Discard Changes** confirmation is not displayed when an admin adds people on the **System Roles** System Console page and clicks elsewhere before saving the changes.
```
Comment (cwarnermm) [inline source/administration/changelog.md:54]:
```suggestion
 - **Discard Changes** confirmation is not displayed when a System Administrator adds people on the **System Roles** System Console page and clicks elsewhere before saving the changes.
```

File: source/administration/changelog.md [hunk 10]
```diff
@@ -4,6 +4,72 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Error text is missing when the team name is left blank on the team creation page.
+ - Posts created by bots containing attachments sometimes appear as repeated until the user refreshes the page.
+ - Emoji counter in the center channel doesn't always update immediately when a reaction is added in the right-hand side.
+ - A JavaScript error may appear in some cases when dismissing the new messages toast while scrolled up in the right-hand side.
+ - The Admin Filter option is not disabled on the AD/LDAP page for admin roles with the ``sysconsole_write_authentication`` permission.
```
Comment (cwarnermm) [inline source/administration/changelog.md:59]:
```suggestion
 - The **Admin Filter** option is not disabled on the AD/LDAP page for admin roles with the ``sysconsole_write_authentication`` permission.
```

File: source/administration/changelog.md [hunk 11]
```diff
@@ -4,6 +4,72 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - On a server using a subpath, the URL opens a blank page if the System Admin changes the Site URL in the System Console. To fix this, the System Admin should restart the server.
```
Comment (cwarnermm) [inline source/administration/changelog.md:60]:
```suggestion
 - On a server using a subpath, the URL opens a blank page if the System Administrator changes the Site URL in the System Console. To fix this, the System Admin should restart the server.
```

File: source/administration/changelog.md [hunk 12]
```diff
@@ -4,6 +4,72 @@ This changelog summarizes updates to [Mattermost Team Edition](https://mattermos
...
+ - Login does not work when Custom Terms of Service is enabled and MFA is enforced.
+ - Google login fails on the Classic mobile apps.
+ - Status may sometimes get stuck as **Away** or **Offline** in High Availability mode with IP Hash turned off.
+ - Searching stop words in quotes with Elasticsearch enabled returns more than just the searched terms.
```
Comment (cwarnermm) [inline source/administration/changelog.md:64]:
```suggestion
 - Searching stop words in quotation marks with Elasticsearch enabled returns more than just the searched terms.
```

**Review by cwarnermm — APPROVED** [review]
Very minor suggestions inline. Reads very well!

**Pull request merged**